### ISSUE TITLE
Set custom headers in info.json request

### ISSUE THREAD

**Issue by gmatto** [ProjectMirador/mirador#3081]
Here is my config settings that allow to add custom headers to the manifest and images requests:
```
config = {
    // ...
    resourceHeaders: {
        'Authorization': `Bearer ${token}`
    },
    osdConfig: {
        loadTilesWithAjax: true,
        ajaxHeaders: {
            'Authorization': `Bearer ${token}`
        }
    }
}
```
But it seems there is no way to include authorization header in `info.json` request.
Also, I tried to use cookie, but my image server is located on a different domain. Parameter `credentials` of the `fetch` method inside function `fetchInfoResponse` is set to `same-origin` value by default, so cookie is not sent in this case.

_P.S._ The manifest is based on `IIIF Presentation API 2.1.1` model. `IIIF Authentication API` is not used.

**Comment by mejackreed** [comment]
Does using `ajaxWithCredentials` or setting the `crossOriginPolicy` in the osdConfig help?

**Comment by gmatto** [comment]
> Does using `ajaxWithCredentials` or setting the `crossOriginPolicy` in the osdConfig help?

No, that didn’t help. `osdConfig` settings apply to image requests only, not `info.json` request.

**Comment by cbeer** [comment]
I wondered about something similar when I was recently working on this code. It's unclear to me if `resourceHeaders` provides the right interface for us, but I'm also not entirely clear what the use cases for it are.

In this case, it seems like you're implementing something like the IIIF auth spec, but are providing the access token directly instead of using a service. I imagine you'd only want to send your token on requests to particular providers.

I can imagine other cases where additional headers are needed to work around implementation  issues (say, forcing IIIF v2 or v3 for servers that do content negotiation), or providing analytics or usage statistics.

Maybe a better configuration interface for this might be a hash of regular expressions to check against the request url, or more generically functions that receive the request and decide if it  applies?

**Comment by mejackreed** [comment]
The initial use case for `resourceHeaders` was the v2 / v3 content negotiation use case. This really only applied to manifests, but definitely agree that the configuration interface could be enhanced.

I think the regex or function based approach would work here to accommodate these cases.

**Comment by cbeer** [comment]
👍 ; even in content negotiation, I'm not sure you'd really want to negotiate for one or the other for all  providers.

**Comment by mejackreed** [comment]
Agree.. the regex or function approach would be preferred for sure. This also sparks the notion that i've seen elsewhere where we could use a similar approach (regex/function) to post process the responses. @jbaiter and others are doing something similar in plugins already.

**Comment by cbeer** [comment]
I believe our saga support should nicely fit the post-processing niche, but I'd be interested in seeing @jbaiter 's  work.

**Comment by mejackreed** [comment]
I think they are doing something similar to what is described here: https://github.com/ProjectMirador/mirador/issues/2971#issuecomment-631479528

**Comment by mejackreed** [comment]
Resolved in #3082
